packages/schema/tests/idl.rs

```diff
@@ -136,13 +136,96 @@ fn test_query_responses_generics() {
         .as_array()
         .unwrap();
 
-    // Find the "balance" query in the queries schema
+    // Find the "query_data" query in the queries schema
     assert_eq!(queries.len(), 1);
     assert_eq!(
         queries[0].get("required").unwrap().get(0).unwrap(),
         "query_data"
     );
 
-    // Find the "balance" query in responses
+    // Find the "query_data" query in responses
     api.get("responses").unwrap().get("query_data").unwrap();
 }
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema, QueryResponses)]
+#[serde(untagged)]
+#[query_responses(nested)]
+pub enum NestedQueryMsg {
+    Query(QueryMsg),
+    Sub(SubQueryMsg1),
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema, QueryResponses)]
+#[serde(rename_all = "snake_case")]
+pub enum SubQueryMsg1 {
+    #[returns(u128)]
+    Variant1 { test: String },
+}
+
+#[test]
+fn test_nested_query_responses() {
+    let api_str = generate_api! {
+        instantiate: InstantiateMsg,
+        query: NestedQueryMsg,
+    }
+    .render()
+    .to_string()
+    .unwrap();
+
+    let api: Value = serde_json::from_str(&api_str).unwrap();
+    let queries = api
+        .get("query")
+        .unwrap()
+        .get("anyOf")
+        .unwrap()
+        .as_array()
+        .unwrap();
+    let definitions = api.get("query").unwrap().get("definitions").unwrap();
+
+    // Find the subqueries
+    assert_eq!(queries.len(), 2);
+    assert_eq!(
+        queries[0].get("$ref").unwrap().as_str().unwrap(),
```

Comment:
Ah, I see you had the added challenge of figuring the schema structure out. Nice!